### PR TITLE
Fixes to links and link anchors

### PR DESCRIPTION
**Pull request by hamishwillee**
- Fixes a bunch of link issues. 
- Adds a basic link checker for vuepress: https://github.com/ulivz/vuepress-plugin-check-md (wouldn't just use this, but helpful "as well")
- Increases search values returned to 10 (from 5) and adds version to search params

### CODE REVIEW

File: en/advanced_config/sensor_thermal_calibration.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ To perform an offboard calibration:
 
 1. Ensure the frame type is set before calibration, otherwise calibration parameters will be lost when the board is setup.
 1. Power up the board and set the [TC_A_ENABLE](../advanced_config/parameter_reference.md#TC_A_ENABLE), [TC_B_ENABLE](../advanced_config/parameter_reference.md#TC_B_ENABLE) and [TC_G_ENABLE](../advanced_config/parameter_reference.md#TC_G_ENABLE) parameters to `1`.
-1. Set all [CAL_GYRO*](../advanced_config/parameter_reference.md#CAL_GYRO0_EN) and [CAL_ACC*](../advanced_config/parameter_reference.md#CAL_ACC0_EN) parameters to defaults.
+1. Set all [CAL_GYRO*](../advanced_config/parameter_reference.md#CAL_GYRO0_ID) and [CAL_ACC*](../advanced_config/parameter_reference.md#CAL_ACC0_ID) parameters to defaults.
```
Comment (hamishwillee) [inline en/advanced_config/sensor_thermal_calibration.md:81]: @bkueng FYI, these params changed names at some point - ie CAL_GYRO0_EN no longer exists. I have assumed. assume that the instruction to just set the CAL_GRYO and CAL_ACC is still fine though.

File: en/advanced_config/esc_calibration.md [hunk 2]
```diff
@@ -48,6 +48,6 @@ To calibrate the ESCs:
 
 :::note
 High-quality controllers come with a factory calibration. 
-In *theory* this means that they can be configured by just setting the [PWM_MIN](../advanced_config/parameter_reference.md#PWM_MIN) and [PWM_MAX](../advanced_config/parameter_reference.md#PWM_MAX) parameters to the values provided in the ESC technical specification.
+In *theory* this means that they can be configured by just setting the [PWM_MIN](../advanced_config/parameter_reference.md#PWM_MIN) and [PWM_MAX](../advanced_config/parameter_reference.md#PWM_MAX) parameters to the values provided in the ESC technical specification.
```
Comment (hamishwillee) [inline en/advanced_config/esc_calibration.md:51]: @bkueng PWM_MIN and PWM_MIN are no longer in the parameters. Do you know what I can replace them with here?

**Comment by hamishwillee** [comment]
I am merging this, but @bkueng would appreciate your answer to https://github.com/PX4/PX4-user_guide/pull/1303#discussion_r644542052 when you have a chance :-)

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 3]
```diff
@@ -211,7 +211,7 @@ Once the vehicle has performed sufficient horizontal movement to make the yaw ob
 
 Data from GPS receivers can be blended using an algorithm that weights data based on reported accuracy (this works best if both receivers output data at the same rate and use the same accuracy).
 The mechanism also provides automatic failover if data from a receiver is lost (it allows, for example, a standard GPS to be used as a backup to a more accurate RTK receiver).
-This is controlled by the [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) parameter.
+This is controlled by the [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) parameter.
```
Comment (hamishwillee) [inline en/advanced_config/tuning_the_ecl_ekf.md:214]: @bkueng EKF2_GPS_MASK also appears to have disappeared. Do you know what has replaced it? (and whether I can just plug that in here, or it needs bigger change?)

File: en/advanced_config/esc_calibration.md (hunk 2 above)
Comment (bkueng) [reply en/advanced_config/esc_calibration.md:51]: PWM_MAIN_MAX/PWM_AUX_MAX

File: en/advanced_config/tuning_the_ecl_ekf.md (hunk 3 above)
Comment (bkueng) [reply en/advanced_config/tuning_the_ecl_ekf.md:214]: I don't know about that one

**Pull request merged**